pkg/operator/resourcesynccontroller/resourcesync_controller.go

```diff
@@ -124,10 +129,14 @@ func (c *ResourceSyncController) SyncPartialConfigMap(destination ResourceLocati
 }
 
 func (c *ResourceSyncController) SyncSecret(destination, source ResourceLocation) error {
-	return c.SyncPartialSecret(destination, source)
+	return c.SyncPartialSecret(destination, source, noPreconditions)
+}
+
+func (c *ResourceSyncController) SyncSecretConditionally(destination, source ResourceLocation, preconditionsFulfilledFn PreconditionsFulfilled) error {
+	return c.SyncPartialSecret(destination, source, preconditionsFulfilledFn)
 }
 
-func (c *ResourceSyncController) SyncPartialSecret(destination, source ResourceLocation, keys ...string) error {
+func (c *ResourceSyncController) SyncPartialSecret(destination, source ResourceLocation, preconditionsFulfilledFn PreconditionsFulfilled, keys ...string) error {
```

Comment:
this is a breaking change, wondering if there is a way to avoid it?

Reply:
Good catch, I change that and kept the initial signature of the function and used an intermediate function to do the computation: 
https://github.com/openshift/library-go/pull/1243/files#diff-c72f483f7c5a6fb8b8c216cb67361320b8c81865eede67417512622c452f8b9eR114